src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -188,7 +184,13 @@ def get_experiment_results(self, qi_job):
             histogram = QuantumInspireBackend.__convert_histogram(result, measurements, result['number_of_qubits'],
                                                                   job['number_of_shots'])
             histogram_obj = Obj.from_dict(histogram)
-            experiment_result_data = ExperimentResultData(counts=histogram_obj)
+            raw_data_url = result.get('raw_data_url')
+            raw_data = self.__api.get(raw_data_url)
+            memory_data = self.__get_memory_data(raw_data, measurements, result['number_of_qubits'])
+            if memory_data:
```

Comment:
Why not always add the `memory_data `to the `experiment_result_data`? I think it is save to assume that if there is a histogram there also is some raw data.

Reply:
There is one exception. When number of shots = 1 there is no memory data (raw_data). In this case there is a histogram. I added a unittest for this special case.
The if is necessary because ExperimentResultData won't accept None as memory parameter.